src/views/open-api-docs-view-v2/utils/group-items-by-key.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * Given a flat array of items, and a getItemGroupKey function,
+ * Return an array of item groups, with item groups being constructed
+ * based on keys derived from the provided getItemGroupKey function.
+ */
+export function groupItemsByKey<Item>(
```

Comment:
I had a think through approaches to grouping operation objects, and found my way to this pretty generic function. This gets used with `getOperationGroupKey` passed as the `getItemGroupKey` function in order to group operations using arbitrary logic.